model-engine/tests/unit/inference/test_vllm_batch.py

```diff
@@ -0,0 +1,308 @@
+@patch(
+    "model_engine_server.inference.batch_inference.vllm_batch.CreateBatchCompletionsRequestContent"
+)
+@patch("model_engine_server.inference.batch_inference.vllm_batch.generate_with_vllm")
+@patch("model_engine_server.inference.batch_inference.vllm_batch.get_s3_client")
+@patch("subprocess.Popen")
+@patch(
+    "model_engine_server.inference.batch_inference.vllm_batch.smart_open.open",
+    new_callable=mock_open,
+    read_data="Mocked content",
+)
+async def test_batch_inference_failed_to_download_model(
+    mock_open_func,
+    mock_popen,
+    mock_get_s3_client,
+    mock_generate_with_vllm,
+    mock_create_batch_completions_request_content,
+    mock_create_batch_completions_request,
+    create_batch_completions_request,
+    create_batch_completions_request_content,
+    create_vllm_request_outputs,
+    mock_s3_client,
+    mock_process,
+):
+    # Mock the necessary objects and data
+    mock_process.returncode = 1
+    mock_popen.return_value = mock_process
+    mock_get_s3_client.return_value = mock_s3_client
+    mock_create_batch_completions_request.parse_file.return_value = create_batch_completions_request
+    mock_create_batch_completions_request_content.parse_raw.return_value = (
+        create_batch_completions_request_content
+    )
+
+    # Call the function
+    with pytest.raises(IOError):
+        await batch_inference()
+
+
+@pytest.mark.asyncio
+@patch("model_engine_server.inference.batch_inference.vllm_batch.CreateBatchCompletionsRequest")
+@patch(
+    "model_engine_server.inference.batch_inference.vllm_batch.CreateBatchCompletionsRequestContent"
+)
+@patch("model_engine_server.inference.batch_inference.vllm_batch.generate_with_vllm")
+@patch("model_engine_server.inference.batch_inference.vllm_batch.get_s3_client")
+@patch("subprocess.Popen")
+@patch(
+    "model_engine_server.inference.batch_inference.vllm_batch.smart_open.open",
+    new_callable=mock_open,
+    read_data="Mocked content",
+)
+@patch("model_engine_server.inference.batch_inference.vllm_batch.os.getenv")
+async def test_batch_inference_two_workers(
+    mock_getenv,
+    mock_open_func,
+    mock_popen,
+    mock_get_s3_client,
+    mock_generate_with_vllm,
+    mock_create_batch_completions_request_content,
+    mock_create_batch_completions_request,
+    create_batch_completions_request,
+    create_batch_completions_request_content,
+    create_vllm_request_outputs,
+    mock_s3_client,
+    mock_process,
+):
+    # Mock the necessary objects and data
+    mock_popen.return_value = mock_process
+    mock_get_s3_client.return_value = mock_s3_client
+    create_batch_completions_request.data_parallelism = 2
+    mock_create_batch_completions_request.parse_file.return_value = create_batch_completions_request
+    mock_create_batch_completions_request_content.parse_raw.return_value = (
+        create_batch_completions_request_content
+    )
+
+    mock_results_generator = MagicMock()
+
+    mock_completion_output = CompletionOutput(
```

Comment:
nit: I think this can be defined via fixture as well just so the mocked input / output code is closer together